cmd/dao_tests/account_test.go

```diff
@@ -0,0 +1,169 @@
+//go:build integration
+// +build integration
+
+package main
+
+import (
+	"context"
+	"database/sql"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	_ "github.com/RHEnVision/provisioning-backend/internal/dao/sqlx"
+	"github.com/RHEnVision/provisioning-backend/internal/models"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func createAccount() *models.Account {
+	return &models.Account{
+		OrgID:         "2",
+		AccountNumber: sql.NullString{String: "100", Valid: true},
+	}
+}
+
+func createAccountWithNullAccountNumber() *models.Account {
+	return &models.Account{
+		OrgID:         "2",
+		AccountNumber: sql.NullString{},
+	}
+}
+
+func setupAccount(t *testing.T) (dao.AccountDao, context.Context) {
+	setup()
+	ctx := identity.WithTenant(t, context.Background())
+	accDao, err := dao.GetAccountDao(ctx)
+	if err != nil {
+		panic(err)
+	}
+	return accDao, ctx
+}
+
+func teardownAccount(_ *testing.T) {
+	teardown()
+}
+
+func TestCreateAccount(t *testing.T) {
+	accDao, ctx := setupAccount(t)
+	defer teardownAccount(t)
+	acc := createAccount()
+	err := accDao.Create(ctx, acc)
+	if err != nil {
+		t.Errorf("create failed: %v", err)
+		return
+	}
+	account, err := accDao.GetById(ctx, 2)
+	if err != nil {
+		t.Errorf("getById failed: %v", err)
+		return
+	}
+
+	assert.Equal(t, acc.OrgID, account.OrgID, "Create account error: OrgID does not match.")
+	assert.Equal(t, acc.AccountNumber.String, account.AccountNumber.String, "Create account error: AccountNumber does not match.")
+	assert.Equal(t, acc.AccountNumber.Valid, account.AccountNumber.Valid, "Create account error: AccountNumber does not match.")
+}
+
+func TestCreateAccountWithNullAccountNumber(t *testing.T) {
+	accDao, ctx := setupAccount(t)
+	defer teardownAccount(t)
+	acc := createAccountWithNullAccountNumber()
+	err := accDao.Create(ctx, acc)
+	if err != nil {
+		t.Errorf("create failed: %v", err)
+		return
+	}
+	account, err := accDao.GetById(ctx, 2)
+	if err != nil {
+		t.Errorf("getById failed: %v", err)
+		return
+	}
+
+	assert.Equal(t, acc.OrgID, account.OrgID, "CreateAccountWithNullAccountNumber error: account OrgID does not match.")
+	assert.Equal(t, acc.AccountNumber.String, account.AccountNumber.String, "CreateAccountWithNullAccountNumber error: AccountNumber does not match.")
+	assert.Equal(t, acc.AccountNumber.Valid, account.AccountNumber.Valid, "CreateAccountWithNullAccountNumber error: AccountNumber does not match.")
```

Comment:
The same case, when it's obvious these values must be equal, you simply skip the messages:

```suggestion
	assert.Equal(t, acc.AccountNumber.Valid, account.AccountNumber.Valid)
```

I would only use it for example in the Pubkey.Fingerprint field which is auto generated and there it makes sense to explain what the value should contain.

I am just trying to save you a lot of typing and possibly ease burden of maintaining those messages (e.g. after rename of field etc).